keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1800 @@
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Re-enabling the feature gate and restarting the Kubelet will restore the
+pod-level resource management capabilities. The Kubelet will once again be able
+to admit and manage pods according to the `pod` and enhanced `container` scope
+logic.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Unit tests will be added to verify that the Kubelet's resource and topology
+managers correctly handle the feature gate being enabled or disabled.
+Specifically, tests will cover the conditional logic in the `Admit` and
+`Allocate` functions.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+A rollout could fail if a bug in the new allocation logic prevents `Guaranteed`
+pods from starting. This would not impact already running workloads, but it
+would prevent new `Guaranteed` pods from being aligned and allocated on the
+affected node. A rollback (disabling the feature gate and restarting the
+Kubelet) would mitigate this for new pods.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+A significant increase in the following metrics after enabling the feature would
+be a strong indicator that a rollback is needed:
+
+-   `topology_manager_admission_errors_total`: Indicates that the Topology
+    Manager is rejecting pods that it would have previously admitted.
+-   `cpu_manager_pinning_errors_total`: Indicates failures in the CPU allocation
+    logic.
+-   `memory_manager_pinning_errors_total`: Indicates failures in the Memory
+    allocation logic.
```

Comment:
let's think if we need new metrics. For example pods with exclusive resources allocation and containers with exclusive resource allocation? just thinking aloud